indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1726,6 +1726,10 @@ private void discoverTasks() throws ExecutionException, InterruptedException, Ti
     List<String> futureTaskIds = new ArrayList<>();
     List<ListenableFuture<Boolean>> futures = new ArrayList<>();
     List<Task> tasks = taskStorage.getActiveTasksByDatasource(dataSource);
+    final Map<String, Task> activeTaskMap = new HashMap<>();
+    for (Task task : tasks) {
+      activeTaskMap.put(task.getId(), task);
+    }
```

Comment:
Maybe put this whole part in a separate method `getActiveTaskMap` as it is used in two places.

Reply:
Done